pkg/apis/serving/metadata_validation.go

```diff
@@ -41,9 +41,16 @@ var (
 
 // ValidateObjectMetadata validates that `metadata` stanza of the
 // resources is correct.
-func ValidateObjectMetadata(meta metav1.Object) *apis.FieldError {
+func ValidateObjectMetadata(ctx context.Context, meta metav1.Object) *apis.FieldError {
+	allowZeroInitialScale := false
+	if config.FromContext(ctx) != nil {
+		autoscalerConfig := config.FromContext(ctx).Autoscaler
+		if autoscalerConfig != nil {
+			allowZeroInitialScale = autoscalerConfig.AllowZeroInitialScale
+		}
+	}
 	return apis.ValidateObjectMetadata(meta).
-		Also(autoscaling.ValidateAnnotations(meta.GetAnnotations()).
+		Also(autoscaling.ValidateAnnotations(allowZeroInitialScale, meta.GetAnnotations()).
```

Comment:
Why isn't this just passing in context and moving the above into the autoscaling validation?

Reply:
There's a cyclic package import problem if we do so since we have to import `knative.dev/serving/pkg/apis/config` in `knative.dev/serving/pkg/apis/autoscaling` to get the configmap.
```
can't load package: import cycle not allowed
package knative.dev/serving/pkg/apis/autoscaling
	imports knative.dev/serving/pkg/apis/config
	imports knative.dev/serving/pkg/autoscaler/config
	imports knative.dev/serving/pkg/apis/autoscaling
```
Is there any workaround for this problem?